### DISCUSSION TITLE
taro小程序从3.0.14升级到3.3.14，路由上的__key_不见了？

### DISCUSSION THREAD

**Discussion by theinkoftheink** [NervJS/taro#10676] [Q & A] [answered]
taro小程序从3.0.14升级到3.3.14，路由上的__key_不见了？
路由参数里面多了一个 $taroTimestamp 这个是替换 __key_使用的吗？

**Comment by Chen-jj** [comment]
`$taroTimestamp` 是页面的唯一标识，应该可以满足使用要求

**Reply by theinkoftheink** [reply]
可以的，但是不清楚后面还会不会再修改成其他标识，目前代码里面用了__key_升级后要全部替换，比较担心这个标识的稳定性如何

**Comment by Chen-jj** [comment, marked as answer]
`__key__` 也不是一个公开的 API 吧，我刚找了一下都没找到之前在哪里设置的。。`$taroTimestamp` 是内部用来标记页面唯一的，是打开页面时的 `Date.now()`，没什么必要基本不会动它。

**Reply by Chen-jj** [reply]
如果是要使用页面的唯一标识的话，这个比较稳：https://github.com/NervJS/taro/blob/96d093cb378d38e23e0665c43670eb2fa1aa7e67/packages/taro-runtime/src/dsl/common.ts#L111

**Reply by theinkoftheink** [reply]
好的，谢谢啦

**Reply by neo693** [reply]
如果不加这个$taroTimestamp 会有什么问题吗

**Reply by neo693** [reply]
<img width="691" alt="截屏2022-02-28 下午4 01 04" src="https://user-images.githubusercontent.com/28082583/155946205-a23a4e4f-bdb5-4ba8-8ac8-e48e2282165c.png">
百度这边会有这个问题 影响seo 本来就一个页面 结果变成了几万个 删除死链的时候就特别麻烦
